Systemd service no block handling for salt-minion service

barneysowood opened the pull request:
### What does this PR do?

Modifies `systemd_service.{restart,stop}` to default to using `no_block=True` when the service being stopped or restarted is the salt-minion.

If you don't pass `no_block=True`, the minion blocks waiting for systemd to restart the service, while systemd is waiting for the minion to exit. Eventually, after systemd hits its timeout (defaulting to 90secs) it will kill the salt minion processes.

Behaviour for other services should remain the same and the functions will still honour the value of `no_block` if passed as an argument.

When combined with #68209 this should lead to a much better default behaviour when calling `service.restart` on the salt-minion service.

### Previous Behavior
When calling `service.restart` or `service.stop` on the salt-minion service, you would have to add `no_block=True` otherwise the salt-minion service and system would deadlock until systemd eventually killed the minion processes.

### New Behavior
If called on the salt-minion service, `service.restart` and `service.stop` default to `no_block=True`. For other services, default remains `no_block=False`

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with SSH key?
Yes

<!-- Please review Salt's Contributing Guide for best practices and guidance in
choosing the right branch:
https://docs.saltproject.io/en/master/topics/development/contributing.html -->

<!-- Additional guidance for pull requests can be found here:
https://docs.saltproject.io/en/master/topics/development/pull_requests.html -->

<!-- See GitHub's page on GPG signing for more information about signing commits
with GPG:
https://help.github.com/articles/signing-commits-using-gpg/ -->

barneysowood committed:
Updated docstrings to reflect the correct action

barneysowood committed:
Modify systemd_service to use no_block for minion

Modifies systemd_service.{restart,stop} to default to using
no_block=True when the service being stopped or restarted is the
salt-minion.

If you don't pass no_block, the minion blocks waiting for systemd to
restart the service, while systemd is waiting for the minion to exit.
Eventually, after systemd hits its timeout it will kill the salt minion
processes.

Behaviour for other services should remain the same and the functions
will still honour the value of no_block if passed as an argument.

barneysowood committed:
Add changelog

twangboy commented:
Looks like some related test failures on Amazon Linux

dwoz commented:
@barneysowood Our amazonlinux 2 test containers do not have a fully functional systemd. Please skip the failing tests and we'll get this one merged.

barneysowood committed:
Skip systemd test on Amazon Linux 2

Skips the test_operation_no_block_default test on Amazon Linux 2 as the
CI conatiner for that distro doesn't have a complete systemd setup.
